test_util/dcos_api_session.py

```diff
@@ -30,11 +30,111 @@ def get_args_from_env():
         'public_masters': os.environ['PUBLIC_MASTER_HOSTS'].split(','),
         'slaves': os.environ['SLAVE_HOSTS'].split(','),
         'public_slaves': os.environ['PUBLIC_SLAVE_HOSTS'].split(','),
-        'default_os_user': os.getenv('DCOS_DEFAULT_OS_USER', 'root'),
-        'ca_cert_path': os.getenv('DCOS_CA_CERT_PATH', None)}
+        'default_os_user': os.getenv('DCOS_DEFAULT_OS_USER', 'root')}
+
+
+class DcosUser:
+    """A lightweight user representation for grabbing the auth info and stashing it"""
+    def __init__(self, credentials: dict):
+        self.credentials = credentials
+        self.auth_token = None
+        self.auth_cookie = None
+        self.auth = None
+
+    def authenticate(self, dcos_url: str):
+        logging.info('Attempting authentication')
+        # explicitly use a session with no user authentication for requesting auth headers
+        r = requests.post(path_join(dcos_url, '/acs/api/v1/auth/login'), json=self.credentials, auth=None)
+        r.raise_for_status()
+        logging.info('Received authentication blob: {}'.format(r.json()))
+        self.auth_token = r.json()['token']
+        self.auth_cookie = r.cookies['dcos-acs-auth-cookie']
+        logging.info('Authentication successful')
```

Comment:
getting a 200 back from the post is actually the indicator that authentication was successful. In the off-chance that we get a successful error code, but a broken payload, I wait until we have what is necessary to auth other requests until formally logging that Authentication is successful. Actual prepping of requests is done by attaching DcosAuth to DcosApiSession.session